### PR TITLE
Custom LinkedList task

### PR DESCRIPTION
**Pull request by desckapg**
### Linked List Implementation

* Added `LinkedList<T>` class implementing a generic, non-thread-safe doubly linked list with O(1) add/remove operations at both ends, strict index bounds checking, and support for null elements.
* Provided a detailed `README.md` documenting the API, usage examples, complexity, and build instructions.

### Build and Test Setup

* Added Gradle build configuration (`build.gradle.kts`) with dependencies for JUnit 5 and AssertJ, and enabled JaCoCo for test coverage.
* Registered the `linkedlist` module in the project settings.

### Testing

* Implemented a comprehensive test suite (`LinkedListTest.java`) covering all public API methods, edge cases, and exception scenarios.

### CODE REVIEW

File: linkedlist/src/main/java/by/desckapg/linkedlist/LinkedList.java [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+     * @return the value of the removed first element
+     * @throws NoSuchElementException if the list is empty
+     */
+    public T removeFirst() {
+        if (head == null) {
+            throw new NoSuchElementException();
+        }
+        var value = head.value;
+        head = head.next;
+        head.prev = null;
```
Comment (maxplyushko) [inline linkedlist/src/main/java/by/desckapg/linkedlist/LinkedList.java]: After head = head.next;, the new head might be null (if the list had only one element).
In that case, head.prev = null; will throw a NullPointerException.

File: linkedlist/src/main/java/by/desckapg/linkedlist/LinkedList.java [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+     * @return the value of the removed last element
+     * @throws NoSuchElementException if the list is empty
+     */
+    public T removeLast() {
+        if (tail == null) {
+            throw new NoSuchElementException();
+        }
+        var value = tail.value;
+        tail = tail.prev;
+        tail.next = null;
```
Comment (maxplyushko) [inline linkedlist/src/main/java/by/desckapg/linkedlist/LinkedList.java]: same as 156

File: linkedlist/src/main/java/by/desckapg/linkedlist/LinkedList.java [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+        } else if (index == size - 1) {
+            return removeLast();
+        } else {
+            Node<T> slow = null;
+            Node<T> fast = head;
+            for (; index > 0 && fast != null; index--) {
+                slow = fast;
+                fast = fast.next;
+            }
+            slow.next = fast.next;
```
Comment (maxplyushko) [inline linkedlist/src/main/java/by/desckapg/linkedlist/LinkedList.java:208]: it updates slow.next = fast.next but doesn't update the previous reference of the next node. This breaks the doubly-linked structure.

File: linkedlist/src/main/java/by/desckapg/linkedlist/LinkedList.java [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+     * Note: valid indices are [0, size-1]; to append use addLast(value).
+     *
+     * @param index index before which the specified element is to be inserted (0..size-1)
+     * @param value element to be inserted
+     * @throws IndexOutOfBoundsException if the index is out of range
+     */
+    public void add(int index, T value) {
+        if (index == 0) {
+            addFirst(value);
+        } else if (index == size) {
```
Comment (maxplyushko) [inline linkedlist/src/main/java/by/desckapg/linkedlist/LinkedList.java]: The JavaDoc says valid indices are [0, size-1], but the implementation allows index == size. This is inconsistent with the documentation.

File: linkedlist/src/main/java/by/desckapg/linkedlist/LinkedList.java [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+    }
+
+    /**
+     * Returns the value at the specified position in this list.
+     *
+     * @param index index of the element to return (0..size-1)
+     * @return the value at the specified position
+     * @throws IndexOutOfBoundsException if the index is out of range
+     */
+    public T get(int index) {
```
Comment (maxplyushko) [inline linkedlist/src/main/java/by/desckapg/linkedlist/LinkedList.java:69]: Missing null check in get(). The loop in the get(int index) method doesn't properly handle the case where node becomes null during traversal.

Comment (desckapg) [reply linkedlist/src/main/java/by/desckapg/linkedlist/LinkedList.java:69]: But if linked list has correct doubly structure it's unreachable case when node becomes null while traversing in [0, size-1] range?

**Review by maxplyushko — APPROVED** [review]

File: linkedlist/src/main/java/by/desckapg/linkedlist/LinkedList.java (hunk 5 above)
Comment (maxplyushko) [reply linkedlist/src/main/java/by/desckapg/linkedlist/LinkedList.java:69]: agree
